### PR TITLE
Add Unit tests in sdo build

### PR DESCRIPTION
**Pull request by nbmaiti**
This PR add framework and source of test framework and unit test code for sdo.
supported platform - linux 

-> Add unit test based framework based on unity
    https://github.com/ThrowTheSwitch/Unity/tree/v2.5.1

    -> Add unit test files for sdo and hal apis are located inside
    tests/unit/* as test_*.c files

    -> To compile and run unit tests use following commands
	make pristine || true;
	cmake -Dunit-test=true -DHTTPPROXY=true -DBUILD=release \
		-DKEX=ecdh -DAES_MODE=ctr -DDA=ecdsa256 -DPK_ENC=ecdsa . ;
	make

### CODE REVIEW

File: cDevice_Build.sh [hunk 1]
```diff
@@ -53,3 +53,10 @@ cp -a ${WORKSPACE}/data/*.blob ${WORKSPACE}/ecdsa384_c_sct_device_bin/blob_backu
 cp -a ${WORKSPACE}/data/platform_aes_key.bin ${WORKSPACE}/ecdsa384_c_sct_device_bin/blob_backup
 cp -a ${WORKSPACE}/data/platform_hmac_key.bin ${WORKSPACE}/ecdsa384_c_sct_device_bin/blob_backup
 cp -a ${WORKSPACE}/data/platform_iv.bin ${WORKSPACE}/ecdsa384_c_sct_device_bin/blob_backup
+
+echo " *****Running Unit Tests*********" 
+make pristine || true; cmake -Dunit-test=true -DHTTPPROXY=true -DBUILD=release -DKEX=ecdh -DAES_MODE=ctr -DDA=ecdsa256 -DPK_ENC=ecdsa ; make -j4
```
Comment (trbehera) [inline cDevice_Build.sh:58]: Please use -j$(nproc) in place of -j4.

File: docs/linux.md [hunk 2]
```diff
@@ -79,6 +79,16 @@ To run the device against the manufacturer CRI for the DI protocol, do the follo
   $ ./build/linux/${BUILD}/linux-client
   ```
 
+## 7. Compiling and runing of unit tests for SDO
+  Unit-test framework is located inside tests folder.
+
+  use following command to compile and running.
```
Comment (trbehera) [inline docs/linux.md:85]: use => Use

File: tests/unit/test_utils.c [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+	bool bret = true;
+	// no filename
+	bret = file_exists(NULL);
+	TEST_ASSERT_FALSE(bret);
+
+	// filename there but file doesn't exists
+	bret = file_exists("hello.txt");
+	TEST_ASSERT_FALSE(bret);
+
+#if 0
```
Comment (trbehera) [inline tests/unit/test_utils.c:109]: Dead code, possibly can be removed.

Comment (nbmaiti) [reply tests/unit/test_utils.c]: updated

File: cDevice_Build.sh (hunk 1 above)
Comment (nbmaiti) [reply cDevice_Build.sh]: updated

File: docs/linux.md (hunk 2 above)
Comment (nbmaiti) [reply docs/linux.md]: updated

**Pull request merged**